### DISCUSSION TITLE
Facing Issue with jodit editor when template involves foreach loop

### DISCUSSION THREAD

**Discussion by monisha01-art** [xdan/jodit#1198] [Q&A] [answered]
scenario :
 cshtml template will be fetched from sql db and displayed in UI screen using Jodit editor, in the editor , User can edit the template and on clicking on save button, the template will be updated in db.
 Issue: 
 the cshtml template which contains foreach() loop, is completely changed by the editor ,causing failure while using the template on other functionality.
 I have added the snippet and both before and after files , please help.
 Actual template:
 
![image](https://github.com/user-attachments/assets/a04af779-8226-49f6-aaf0-4b7a67e77b4d)
after template is displayed in jodit editor  :(for this one, no changes has been done to the template, we just got template from db, displayed in jodit, and saved)
![image](https://github.com/user-attachments/assets/d0039474-b198-42e6-b81d-c4aab83bfd42)

 
[jodit_template.txt](https://github.com/user-attachments/files/17841083/jodit_template.txt)
[actual_Template.txt](https://github.com/user-attachments/files/17841085/actual_Template.txt)

**Comment by xdan** [comment, marked as answer]
Jodit is an HTML editor — on load it parses your markup into a DOM and re-serializes clean HTML, so server-side template syntax like Razor `@foreach` / `<text>` blocks gets rewritten or dropped. It's not built to round-trip template code, only HTML content.

If people need to edit templates that contain Razor logic, keep the logic outside the editable area (let them edit only the content fragments), or give them the raw source in a plain textarea / Jodit's source mode rather than the WYSIWYG. There's no setting that makes the WYSIWYG preserve arbitrary `@`-code intact.
